pdf_build_src/pandoc_script.py

```diff
@@ -17,19 +18,55 @@ def build_pdf(filename="bids-spec.pdf", logfile="bids-spec_pandoc_log.json"):
     logfile : str
         Name of the log file. Defaults to "bids-spec_pandoc_log.json".
     """
-    # Files that are not supposed to be built into the PDF
-    EXCLUDE = ["./index.md", "./schema/README.md", "./pregh-changes.md"]
-
-    # Get all input files
-    markdown_list = []
-    for root, dirs, files in os.walk("."):
-        for file in files:
-            fpath = os.path.join(root, file)
-            if fpath.endswith(".md") and fpath not in EXCLUDE:
-                markdown_list.append(fpath)
-            elif fpath.endswith("index.md"):
-                # Special role for index.md
-                index_page = fpath
+    # change directory name if running on CI
+    runs_on_CI = True if os.getenv("CIRCLE_BRANCH") else False
+    proj_fname = "bids-specification"
+    if runs_on_CI:
+        proj_fname = "project"
+
+    # location of this file: This is also the working directory when
+    # the pdf is being built using `cd build_pdf_src` and then
+    # `bash build_pdf.sh`
+    root = pathlib.Path(__file__).parent.absolute()
+
+    # Parse content files from mkdocs.yml file
+    def _unpack(sections):
+        """Flatten a list of dicts of lists."""
+        sections_flat = []
+        for i in sections:
+            if isinstance(list(i.values())[0], str):
+                sections_flat.append(i)
+            elif isinstance(list(i.values())[0], list):
+                for j in list(i.values())[0]:
+                    sections_flat.append(j)
+        return sections_flat
+
+    def _has_only_str2str(sections):
+        """Check list of dict has only str to str mappings."""
+        for i in sections:
+            if not isinstance(list(i.values())[0], str):
+                return False
+        return True
```

Comment:
Doing this as a recursive generator seems a little less roundabout, and requires only one application.

```suggestion
    def _flatten_values(lst):
        """Flatten a list of dicts of lists to a list of values."""
        for val in (obj.values() for obj in lst):
            if isinstance(val, str):
                yield val
            else:
                yield from _flatten_values(val)
```